### DISCUSSION TITLE
A way to define a 0 point coordinate and planes in meshroom for photogrammetry?

### DISCUSSION THREAD

**Discussion by aleksas** [alicevision/Meshroom#1533] [Ideas]
Meshroom always generates rotated model, which is perfectly reasonable. Is there a way to tell meshroom where is the 0 point and where the axis should be?

If there is no way to do that now maybe that would be a nice feature: For instance when reconstructing room 3d model maybe selecting a pair feature pints in the scene (probably on the floor/base plane) and providing their 3d coordinates would allow meshroom to scale, translate and rotate the model appropriately.

**Comment by natowi** [comment]
See
https://github.com/alicevision/meshroom/wiki/Model-orientation
https://github.com/alicevision/meshroom/wiki/CCTAG-scaling

**Reply by aleksas** [reply]
Great, thanks. Will try it.

**Comment by aleksas** [comment]
@natowi  I'm having trouble improving CCTAG detection. If single FeatureMatching node is used (SIFT + CCTAG3) then I'm able to scale and rotate model - not precise though (upto 20cm in a ~3m^3 area, with 4 markers placed at least 40cm apart on the floor). If I split FeatureMatching into two nodes as indicated in [https://github.com/alicevision/meshroom/wiki/CCTAG-scaling](https://github.com/alicevision/meshroom/wiki/CCTAG-scaling), then I loose CCTAG features, CCTAG feature matching node output log shows "0 geometric image pair matches:". Correct me if I'm wrong, this is due to geometric validation. It's not obvious how to disable validation. My guess would br passing something to FeatureMatching node "Geometric Filter Type" but what and how is not clear.

**Reply by natowi** [reply]
![grafik](https://github.com/alicevision/Meshroom/assets/8138537/66f0c8da-2df6-4b2b-bbcf-ebdf2f3ab050)

**Comment by aleksas** [comment]
Realized that StructureFromMotion kept "Describer Tyle" link from Feature Matching node that had only Sift features. Also found out the "advanced" attribute checkbox. That didn't fix the 20cm error. Will experiment more with markers.
